File: 453. Minimum Moves to Equal Array Elements.java
// 这是一道数学题，但是可以用来复习array的操作方法:
// 1. sum of array; 2. min element of array; 3. length of array
class Solution {
    public int minMoves(int[] nums) {
        int sum = 0;
        int min = nums[0];

        for (int i : nums) {
            min = Math.min(min, i);
            sum += i;
        }

        return sum - min * nums.length;
    }
}

File: 453. Minimum Moves to Equal Array Elements.py
# if we take m moves to make every number the same:
# target * len(nums) = sum(nums) + m * (len(nums) - 1)
# and 
# target = min(nums) + m
# (regardless which len(nums) - 1 numbers we pick for each round of increment
#  the minimum number will always be part of the pick, so it will always stay
#  as the minimum number, therefore target is m steps away from min(nums))
# so
# sum(nums) - min(nums) * len(nums) = m 
class Solution:
    def minMoves(self, nums: List[int]) -> int:
        return sum(nums) - min(nums) * len(nums)
        